dietpi/dietpi-software

```diff
@@ -8220,7 +8228,71 @@ _EOF_
 		#NextCloud
 		INSTALLING_INDEX=114
 		if (( ${aSOFTWARE_INSTALL_STATE[$INSTALLING_INDEX]} == 1 )); then
+		
+			#If Apache webserver is used, make sure all necessary Apache modules are enabled: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#additional-apache-configurations
+			if (( ${aSOFTWARE_INSTALL_STATE[83]} >= 1 )); then
+				a2enmod rewrite headers env dir mime
+			fi
+			
+			#Make sure all necessary PHP modules are enabled: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
+			if (( $DISTRO == 3 )); then
+				php5enmod ctype curl dom fileinfo gd iconv intl json mbstring posix simplexml xmlwriter xmlreader zip pdo_mysql exif opcache apcu
+			else
+				phpenmod ctype curl dom fileinfo gd iconv intl json mbstring posix simplexml xmlwriter xmlreader zip pdo_mysql exif opcache apcu
+			fi
+			
+			#Create Nextcloud specific webserver + PHP config.
+			#Apache: https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#apache-web-server-configuration
+			if (( ${aSOFTWARE_INSTALL_STATE[83]} >= 1 )); then
+				cat << _EOF_ > /etc/apache2/sites-available/nextcloud.conf
+Alias /nextcloud "/var/www/nextcloud/"
+
+<Directory /var/www/nextcloud/>
+        Options +FollowSymlinks
+        AllowOverride All
+
+        <IfModule mod_dav.c>
+                Dav off
+        </IfModule>
+
+        SetEnv HOME /var/www/nextcloud
+        SetEnv HTTP_HOME /var/www/nextcloud
+
+        #Satisfy Any //This would be necessary in case of server-configured parent folder authentication, but might lead to security issues: https://help.nextcloud.com/t/htaccess-warning-while-configuration-should-be-ok/20280/10?u=michaing
+
+# OPCache configuration: https://docs.nextcloud.com/server/12/admin_manual/configuration_server/server_tuning.html?highlight=opcache#enable-php-opcache
+#php_admin_value opcache.enable 1 //This would produce an error here and is enabled by default anyway: https://support.plesk.com/hc/en-us/articles/115001622189-A-lot-of-errors-in-php-fpm-log-ERROR-Unable-to-set-php-value-opcache-enable-
+php_admin_value opcache.enable_cli 1
+php_admin_value opcache.interned_strings_buffer 8
+php_admin_value opcache.max_accelerated_files 10000
+php_admin_value opcache.memory_consumption 128
```

Comment:
We set OPcache already with: https://github.com/Fourdee/DietPi/blob/master/dietpi/dietpi-software#L7986-L8007 based on RAM on system.

So this entry should be removed.

Reply:
Can check Opcache size with:
http://IP_ADDRESS/opcache.php

Reply:
The thing is that Nextcloud at the moment produces a warning on admin panel, if you choose opcache settings different than this. For that reason I added the settings inside directory directive of apache conf instead of usual php.ini or opcache.ini. That way the cache size is just set locally for the nextcloud folder and for all other web sites/services the global setting from dietpi-software will stay valid.

Reply:
@MichaIng 

> if you choose opcache settings different than this. For that reason I added the settings inside directory directive of apache conf instead of usual php.ini or opcache.ini.

Ok 👍 

But we may have issues with 256MB RAM users and a 128MB cache. Most of our users install multiple software titles, but with the swapfile increase recently, should be fine.

Reply:
I just switched my opcache.memory_consumption to 512MB to see the effect on actual memory usage by htop (and will leave it like this for a while). Actually there didn't change much. As there is a limited amount php files of nextcloud, the actual opcache usage can't grow way larger than ~30MB, even if you install all apps, available at nextcloud appstore. PHP files simply do not grow that large usually (compared to images etc.).* For that reason actual memory overload is highly unlikely.

*This is also the reason why I want to start a discussion on nextcloud side to reduce or even remove that setting from their checks. It simply highly depends on additional usage of the webserver. But as for now, I think we should keep the 128MB, as a warning on nextcloud admin panel will result in reports/questions either here or on help.nextcloud.com.